content/learn/intermediate/outputs-2.md

```diff
@@ -0,0 +1,249 @@
+[prefill]: https://yagpdbembeds.netlify.app/?data=JTdCJTIyZW1iZWQlMjIlM0ElN0IlMjJ0aXRsZSUyMiUzQSUyMlRoaXMlMjBpcyUyMGElMjB0aXRsZSUyMiUyQyUyMmRlc2NyaXB0aW9uJTIyJTNBJTIyVGhpcyUyMGlzJTIwYSUyMGRlc2NyaXB0aW9uLiUyMiUyQyUyMmNvbG9yJTIyJTNBMTY3MTE2ODAlMkMlMjJmaWVsZHMlMjIlM0ElNUIlN0IlMjJuYW1lJTIyJTNBJTIyRmllbGQlMjAxJTIyJTJDJTIydmFsdWUlMjIlM0ElMjJWYWx1ZSUyMDElMjIlMkMlMjJpbmxpbmUlMjIlM0F0cnVlJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMkZpZWxkJTIwMiUyMiUyQyUyMnZhbHVlJTIyJTNBJTIyVmFsdWUlMjAyJTIyJTJDJTIyaW5saW5lJTIyJTNBdHJ1ZSU3RCU1RCUyQyUyMmF1dGhvciUyMiUzQSU3QiUyMm5hbWUlMjIlM0ElMjJsem9kZCUyMiUyQyUyMmljb25fdXJsJTIyJTNBJTIyaHR0cHMlM0ElMkYlMkZjZG4uZGlzY29yZGFwcC5jb20lMkZhdmF0YXJzJTJGNjY1MjQzNDQ5NDA1OTk3MDY2JTJGZDYzMTM3MWJiY2NhMDI3NDgxYTcxMjk1NGUyODQyYmMucG5nJTNGc2l6ZSUzRDI1NiUyMiU3RCU3RCU3RA==
+
+{{< /callout >}}
+
+## Editing Messages
+
+Sending a message is nice and all, but for the sake of keeping things clean, you might want to edit a message instead of
+creating a new one each time something changes. We provide the `editMessage` function for this purpose.
+
+```go
+{{ editMessage channel_id message_id new_message_content }}
+```
+
+The `channel_id` is the ID of the channel in which the message is located. The `message_id` is the ID of the message you
+want to edit. The `new_message_content` is the new content you want to set for the message, completely replacing it.
+You should note that YAGPDB can only edit messages from itself, just like you cannot edit someone else's messages.
+
+For a quick demonstration, consider the following code:
+
+```go
+{{ $messageID := sendMessageRetID nil "Hello, World!" }}
+{{ sleep 5 }}
+{{ editMessage nil $messageID "Goodbye, World!" }}
+```
+
+This code sends a message saying "Hello, World!" and then, after 5 seconds, edits the message to say "Goodbye, World!",
+all in the same channel where the custom command was triggered.
+
+### Editing Embeds
+
+Editing embeds is a little more involved than editing regular messages. You need to provide the entire embed object
+again, old parts included. This is due to how the Discord API functions. The basic idea is that we obtain the old embed
+object, modify it, and then send the modified object back to the `editMessage` function.
+
+A contrived example all within the same custom command looks like the following:
+
+```go
+{{ $embed := cembed
+    "title" "This is a title"
+    "description" "This is a description."
+    "color" 0xff0000
+    "fields" (cslice
+        (sdict "name" "Field 1" "value" "Value 1" "inline" true)
+        (sdict "name" "Field 2" "value" "Value 2" "inline" true)
+    )
+    "author" (sdict
+      "name" .User.Username
+      "icon_url" (.User.AvatarURL "256")
+    )
+}}
+{{ $messageID := sendMessageRetID nil $embed }}
+{{ sleep 5 }}
+
+{{ $new_embed := structToSdict $embed }}
+{{ $new_embed.Set "title" "This is a new title" }}
+{{ $new_embed.Set "description" "This is a new description." }}
+{{ $new_embed.Set "fields" (cslice) }}
```

Comment:
Not sure if I like this. One might rightfully question why `$new_embed.Del "fields"` is insufficient (and the answer would be that `$new_embed.Del "Fields"` is necessary due to `structToSdict` not converting case), but that complicates the explanation considerably. Might be worth leaving this line out and addressing it later in the section discussing deep conversion? If you choose to keep it in, explain why `$new_embed.Del "fields"` doesn't work.